### ISSUE TITLE
Feature request : ifcMaterialLayerSetUsage orderof layers

### ISSUE THREAD

**Issue by R2naizr** [IfcOpenShell/IfcOpenShell#3515]
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/79116991/8a61651f-83f4-49da-b055-cdd2c180e1a7)

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/79116991/49f1c696-ad55-41d1-bc5f-b85b5d291848)

Would it be possible to include a text below the BIMObjectMaterialProperties (above the user created material set) selector which shows the order of material layers in the wall. Maybe  "**From origin**", which gives the user an indication of how they're stacking their materials.

**Comment by Moult** [comment]
The layers are always stacked from the reference line (denoted by the origin) with a potential offset (see OffsetFromReferenceLine in your screenshot) and in the direction defined by DirectionSense (see your screenshot). POSITIVE implies +Y and negative implies -Y for walls.

So yes, the first layer is always the layer closest to the origin and the last layer is the layer furthest away from the origin (in which direction depends on the direction sense).

Would you suggest just some text?

**Comment by R2naizr** [comment]
Yes, just a text to indicate the order. In Revit this would be indicated by Interior and Exterior. Maybe the BB addon could use a simple text "From Origin Point" or something, for new users.

**Comment by Moult** [comment]
That's a great idea! Would you like to have a shot at implementing this and submitting a PR? It's a fairly straightforward addition (I think it also might be nice if the text was centered)

**Comment by R2naizr** [comment]
ok, new to this, but I'll figure out how to do a Pull Request.

**Comment by R2naizr** [comment]
wait does that mean I'll have to put up some code?

**Comment by Moult** [comment]
Indeed :) It's less scary than you think and nowadays things like ChatGPT can guide you through the process and even write the code for you in these simple cases

**Comment by bdamay** [comment]
If i understand this one correctly. I made something that adds the label "From Origin Point" and here what it looks like

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/16347726/0ab131d2-626d-4b75-92d1-df374af6850e)

Though i personnaly feels it doesn't bring much more clarity to the user, I would better like to inform about the layer order on mouse hover may be ? but i don't think it's possible, may be it is ? 
I also didn't find a way to lower line height to bring the label closer to the list

**Comment by R2naizr** [comment]
maybe indicating:

(at the top of the setting)
"Interior layer"

(and a the bottom, after the layer)
"Exterior Layer"

that might provide more clarity

**Comment by Moult** [comment]
IFC itself does not define any axis convention for "Interior" or "Exterior", which is why I'm hesitant to use those terms in the UI until we are absolutely confident that choosing one to be Interior or Exterior is a preferred workflow.

The terms that IFC use are:

 - **Object Placement** (this is the official term for the object's orange origin dot)
 - **Reference Line** (the imaginary 2D line of the wall starting from the **Object Placement**)
 - **Base Line** (the imaginary 2D line of the wall where the first layer begins. This is usually coincident with the Reference Line, but the user may specify an **Offset**).

![ifcmateriallayersetusage_wall-01](https://github.com/IfcOpenShell/IfcOpenShell/assets/88302/b00556c6-9e69-44c8-9d49-bfabc2984723)

IFC does not specify the "other" line, the line where the last layer ends. In our code, we call it the **Side Line**. This is not official at all, just a name I thought had a nice ring to it as well as having the other definition of "[Sideline](https://www.wordnik.com/words/sideline)".

A reminder that layers are not only for walls but also for slabs and coverings and insulation and skirtings and whatever.

With that in mind, feel free to decide what you think provides the most user clarity. I am biased since I've spent too much time in the IFC docs so this really does need a fresh set of eyes.

Whenever you tell me to I'll approve the PR :)

BTW the Blender UI tends to have a convention of Caps Casing All Text.

**Comment by R2naizr** [comment]
Then I guess it would just make sense to state" from origin point"

**Comment by Moult** [comment]
@bdamay it's your call, you're doing the PR :) I've added you to the developers list so you can approve the PR whenever you'd like yourself.

**Comment by bdamay** [comment]
@moult Thank you for providing such useful informations and also adding me in the developers list, very nice of u.
  
Well, you're right I'm not quite satisfied of the way it looks yet as IMO it separates the list a bit too much from the block title. May be a question of taste, but i find the panel already rich enough as it is (IMO). 
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/16347726/46852d5d-6903-4328-a501-4f19832ff9d3)

I would rather leave a description to the operator (as a tooltip) as it is less intrusive to the panel and provide help  in a manner people are usually used to. (Move the mouse and find inline help is what i usually do and it doesn't overload the GUI) 
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/16347726/652ac84e-9ab0-4479-89a1-af114b6f36d8)

What would you think about this @R2naizr ?

**Comment by R2naizr** [comment]
Tooltip sounds good for now. Thank you

**Comment by bdamay** [comment]
Pushed the tooltip option this morning
